plugins/ccgx/fu-ccgx-dock-bb.c

```diff
@@ -0,0 +1,734 @@
+				return FALSE;;
+			}
+
+			if (cyacd_file_info.fw_mode != update_fw_mode ||
+			    (self->fw_primary_update_only == TRUE && self->pd_device_data.fw_mode  == FW_MODE_FW2)) {
+				if (handle_count > 1)
+					continue; /* get next handle */
+				else
+					break;
+			}
+		}
+
+		g_debug ("===== Update FW file Info =====");
+		g_debug ("  Silicon ID : 0x%X", cyacd_file_info.silicon_id);
+		g_debug ("  FW Mode :  FW%u", (guint32)cyacd_file_info.fw_mode);
+
+		if (self->fw_image_type == FW_IMAGE_TYPE_DUAL_SYMMETRIC)
+			g_debug ("  FW IMG :  DUAL SYMMETRIC");
+		else if (self->fw_image_type == FW_IMAGE_TYPE_DUAL_ASYMMETRIC)
+			g_debug ("  FW IMG :  DUAL ASYMMETRIC");
+
+		g_debug ("  Version : %u.%u.%u /  0x%02X(%c%c)", 
+			(guint32)cyacd_file_info.app_version.ver.major,
+			(guint32) cyacd_file_info.app_version.ver.minor,
+			(guint32)cyacd_file_info.app_version.ver.build,
+			cyacd_file_info.app_version.ver.type,
+			((cyacd_file_info.app_version.ver.type>>8) & 0xff),
+			((cyacd_file_info.app_version.ver.type) & 0xff));
+
+		/* enter FW Mode */
+		if (!fu_ccgx_hpi_cmd_enter_flash_mode (device, hpi_handle, error)) {
+			g_warning("enter flash mode error");
+			return FALSE;
+		}
+
+		g_usleep (HPI_CMD_ENTER_FLASH_MODE_DELAY_US); /* wait 10 msec */
+
+		/* set up meta data row */
+		if (update_fw_mode == FW_MODE_FW1)
+			row_num = self->pd_device_data.fw1_meta_row_num;
+		else
+			row_num = self->pd_device_data.fw2_meta_row_num;
+
+		row_data = &row_buffer[4];
+
+		/* read meta data */
+		row_size = self->pd_device_data.fw_row_size;
+
+		fu_device_set_status (device,FWUPD_STATUS_DEVICE_READ);
+
+		if (!fu_ccgx_hpi_cmd_read_flash (device, hpi_handle, row_num, row_data, row_size, error)) {
+			g_warning("fw meta data read error ");
+			return FALSE;
+		}
+
+		/* get meta data */
+		metadata = (CCGxMetaData*)&row_data[self->pd_device_data.fw_meta_offset];
+
+		/* clear meta data Valid */
+		metadata->metadata_valid = 0x00;
+
+		/* write meta data again */
+		fu_device_set_status (device,FWUPD_STATUS_DEVICE_ERASE);
+
+		if (!fu_ccgx_hpi_cmd_write_flash (device, hpi_handle, row_num, row_data, row_size,error)) {
+			g_warning ("fw meta data write error");
+			return FALSE;
+		}
+
+		/* read flash data */
+		fu_device_set_status (device,FWUPD_STATUS_DEVICE_WRITE);
+
+		g_debug ("Writing Firmware...");
+
+		while (fu_ccgx_cyacd_file_read_row (cyacd_handle, row_buffer,CYACD_ROW_BUFFER_SIZE)) {
+			row_num = *((guint16*)(&row_buffer[0]));
+			row_size = *((guint16*)(&row_buffer[2]));
+			row_data = &row_buffer[4];
+
+			/* write flash data */
+			if (!fu_ccgx_hpi_cmd_write_flash (device, hpi_handle, row_num, row_data, row_size, error)) {
+				g_warning ("fw row data write error at %d row", (gint32)row_num);
+				return FALSE;
+			}
+
+			/* update process */
+			fw_pos = fu_ccgx_cyacd_file_get_pos (cyacd_handle);
+			fu_device_set_progress_full ( device, fw_pos, update_fw_size );
```

Comment:
whitespace is funny here

Reply:
OK. I will remove whitespace. 
fu_device_set_progress_full (device, fw_pos, update_fw_size);

Reply:
I fixed it.